src/main/io/asyncfatfs/asyncfatfs.c

```diff
@@ -3252,6 +3252,7 @@ uint32_t afatfs_freadSync(afatfsFilePtr_t file, uint8_t *buffer, uint32_t length
         uint32_t leftToRead = length - bytesRead;
         uint32_t readNow = afatfs_fread(file, buffer, leftToRead);
         bytesRead += readNow;
+        buffer += bytesRead;
```

Comment:
We should advance the buffer by `readNow` (bytes read by the last call to `afatfs_fread`), not by the total bytes read.

Reply:
:facepalm: I changed it at the last minute from adding bytesRead to buffer in the fread call to that and forgot to change the variable